amethyst_renderer/src/pass/flat2d/interleaved.rs

```diff
@@ -394,20 +444,51 @@ impl TextureBatch {
         camera: Option<(&Camera, &GlobalTransform)>,
         sprite_sheet_storage: &AssetStorage<SpriteSheet>,
         tex_storage: &AssetStorage<Texture>,
+        screen_dimensions: &ScreenDimensions,
+    ) {
+        if !self.textures.is_empty() {
+            // Draw to world
+            set_view_args(effect, encoder, camera);
+            TextureBatch::encode_vec(
+                &self.textures,
+                encoder,
+                factory,
+                effect,
+                sprite_sheet_storage,
+                tex_storage,
+            );
+        }
+        if let Some(depth_data) = &effect.data.out_depth {
```

Comment:
can you put this within the `textures_screen` conditional as well? so the depth buffer doesn't clear if it doesn't need to be